pkg/collector/corechecks/snmp/fetch_scalar.go

```diff
@@ -0,0 +1,81 @@
+package snmp
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/gosnmp/gosnmp"
+	"sort"
+	"strings"
+)
+
+func fetchScalarOidsWithBatching(session sessionAPI, oids []string, oidBatchSize int) (scalarResultValuesType, error) {
+	retValues := make(scalarResultValuesType, len(oids))
+
+	batches, err := createStringBatches(oids, oidBatchSize)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create oid batches: %s", err)
+	}
+
+	for _, batchOids := range batches {
+		results, err := fetchScalarOids(session, batchOids)
+		if err != nil {
+			return nil, fmt.Errorf("failed to fetch scalar oids: %s", err.Error())
+		}
+		for k, v := range results {
+			retValues[k] = v
+		}
+	}
+	return retValues, nil
+}
+
+func fetchScalarOids(session sessionAPI, oids []string) (scalarResultValuesType, error) {
+	packet, err := doFetchScalarOids(session, oids)
+	if err != nil {
+		return nil, err
+	}
+	values := resultToScalarValues(packet)
+	retryScalarOids(session, packet, values)
+	return values, nil
+}
+
+// retryScalarOids retries on NoSuchObject or NoSuchInstance for scalar oids not ending with `.0`.
+// This helps keeping compatibility with python implementation.
+// This is not need in normal circumstances where scalar OIDs end with `.0`.
+// If the oid does not end with `.0`, we will retry by appending `.0` to it.
+func retryScalarOids(session sessionAPI, results *gosnmp.SnmpPacket, valuesToUpdate scalarResultValuesType) {
```

Comment:
I'm not sure it will help for readability to have a very long name.
I think the comment might be helpful enough if someone is interested by the implementation details of why we are doing the retry for scalar oids here.

Reply:
alternative to longer name, adding a comment also where you use the function. There's dev friction when using a function with a name that, IMHO, is not explicit, and the explanation is at the function definition position. You ask the reader to step out of the point they are reading to get more details. Personally, when I read it in code I asked why it was retrying with no check if anything failed - I would expect a retry function within an `if failed`, might be just me.

Reply:
Renamed here to `retryFailedScalarOids` https://github.com/DataDog/datadog-agent/pull/6749/commits/0a035fb84fbca850d386f8d2f013b216b3ee18b0

Let me know if that's fine.